Here's the full warning report from the clang-8 compiler that this PR fixes. Please note that since -Werror is on, any compiler warning is going to stall the build (which is what happened in this case).

```
[ 59%] Building CXX object libsolidity/CMakeFiles/solidity.dir/parsing/ParserBase.cpp.o
In file included from /src/solidity/libsolidity/formal/SMTChecker.cpp:18:
In file included from /src/solidity/libsolidity/formal/SMTChecker.h:21:
/src/solidity/libsolidity/formal/SolverInterface.h:65:14: error: explicitly defaulted copy assignment operator is implicitly deleted
      [-Werror,-Wdefaulted-function-deleted]
        Expression& operator=(Expression const&) = default;
                    ^
/src/solidity/libsolidity/formal/SolverInterface.h:174:20: note: copy assignment operator of 'Expression' is implicitly deleted because
      field 'name' has no copy assignment operator
        std::string const name;
                          ^
/src/solidity/libsolidity/formal/SolverInterface.h:66:14: error: explicitly defaulted move assignment operator is implicitly deleted
      [-Werror,-Wdefaulted-function-deleted]
        Expression& operator=(Expression&&) = default;
                    ^
/src/solidity/libsolidity/formal/SolverInterface.h:174:20: note: move assignment operator of 'Expression' is implicitly deleted because
      field 'name' has no move assignment operator
        std::string const name;
                          ^
```

--- a/libsolidity/formal/SolverInterface.h
+++ b/libsolidity/formal/SolverInterface.h
@@ -62,8 +62,6 @@ public:
 
 	Expression(Expression const&) = default;
 	Expression(Expression&&) = default;
-	Expression& operator=(Expression const&) = default;
-	Expression& operator=(Expression&&) = default;
 
 	bool hasCorrectArity() const
 	{

--- a/libsolidity/formal/SolverInterface.h
+++ b/libsolidity/formal/SolverInterface.h
@@ -62,6 +62,8 @@ public:
 
 	Expression(Expression const&) = default;
 	Expression(Expression&&) = default;
+	Expression& operator=(Expression const&) = default;
+	Expression& operator=(Expression&&) = default;
 
 	bool hasCorrectArity() const
 	{
@@ -169,8 +171,8 @@ public:
 		}
 	}
 
-	std::string const name;
-	std::vector<Expression> const arguments;
+	std::string name;
+	std::vector<Expression> arguments;
 	Sort sort;
 
 private:

